app/views/contexts/summary/question.py

```diff
@@ -39,7 +39,16 @@ def __init__(
         )
 
     def _get_answer(self, answer_store, answer_id):
-        return answer_store.get_escaped_answer_value(answer_id, self.list_item_id)
+        if answer_store.get_escaped_answer_value(
+            answer_id, self.list_item_id
+        ) or isinstance(
+            answer_store.get_escaped_answer_value(answer_id, self.list_item_id),
+            int,  # protection against falsy answer value "0"
+        ):
```

Comment:
This should be avoided as we should check for null values. I think we should scrap `get_escaped_answer_value` method, it is only used here. Makes sense if it was used in multiple places, but value source handles it explcity. I would therefore simplify this to something like:

```py
    def _get_answer(self, answer_store, answer_id):
        answer = answer_store.get_answer(
            answer_id, self.list_item_id
        ) or self.schema.get_default_answer(answer_id)

        return escape_answer_value(answer.value) if answer else None
```

If you use walrus, you could do `escape_answer_value` just do non default but it doesn't really matter.

Reply:
So I had to chop it up a little bit otherwise it was never assigning answer the default value cause Answer class instance always exists but not always with value but this code didn't check that, now it is:
```py
def _get_answer(self, answer_store, answer_id):
        answer = answer_store.get_answer(answer_id, self.list_item_id)
        if answer and answer.value:
            return escape_answer_value(answer.value)
        if default_answer := self.schema.get_default_answer(answer_id):
            return default_answer.value
```
I scrapped `get_escaped_answer_value` from answer store as well.

Reply:
`get_answer` will return `None` if the answer didn't exist?. So it will already be handled. Also, you don't want to check the value as value can be falsey. If you get an answer object back, we return the value.

Reply:
I had to add zero check in `_get_answer` as we still need to check for `answer.value` and it evaluates `0` to `False`.

Reply:
If we don't check if `answer.value` is empty string (which evaluates to `False` as you know) then it will never hit `default_value` as it will always pass empty string to the summary.